src/form/formula_gen.cpp

```diff
@@ -309,6 +309,50 @@ bool FormulaGenerator::update(const Operation& op, const RangeMap* ranges) {
       }
       break;
     }
+    case Operation::Type::CLR:
+    case Operation::Type::FIL:
+    case Operation::Type::ROL:
+    case Operation::Type::ROR: {
+      if (op.target.type != Operand::Type::DIRECT || op.source.type != Operand::Type::CONSTANT) {
+        okay = false;
+        break;
+      }
+      int64_t start = op.target.value.asInt();
+      int64_t length = op.source.value.asInt();
+      int64_t left;
+      int64_t right;
+      if (length > 0) {
+        left = start;
+        right = start + length;
+      } else {
+        left = start + length + 1;
+        right = start + 1;
+      }
+      if (right - left >= 15) { // magic number
+        okay = false;
+        break;
+      }
+      if (op.type == Operation::Type::ROL) {
+          auto leftEntry = formula.entries[ExpressionUtil::newFunction(getCellName(left))];
```

Comment:
Add this helper function:
```cpp
Expression FormulaGenerator::cellFunc(int64_t index) const {
  return ExpressionUtil::newFunction(getCellName(index));
}
```
and simplify the code to:

```cpp
auto leftEntry = formula.entries[cellFunc(left))];
```
Do the same to below..